### PR TITLE
[bbs] add pagination to getRepos – EXP-417 EXP-416

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at 54cbf14</samp>

This pull request refactors the `getRepos` method of the `BitbucketServerApi` class to handle different authentication types and pagination. It also updates the methods that use the `getRepos` method in the `BitbucketServerRepositoryProvider` and `BitbucketServerService` classes and adds a test case for the `getRepos` method.

</details>

## Related Issue(s)

Fixes EXP-417 and EXP-416

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-bbs-pagination</li>
	<li><b>🔗 URL</b> - <a href="https://at-bbs-pagination.preview.gitpod-dev.com/workspaces" target="_blank">at-bbs-pagination.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - at-bbs-pagination-gha.15229</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/server/src/bitbucket-server/bitbucket-server-api.ts [hunk 1]
```diff
@@ -283,29 +283,29 @@ export class BitbucketServerApi {
...
+            const pageResult = await this.runQuery<BitbucketServer.Paginated<BitbucketServer.Repository>>(
+                userOrToken,
+                `/repos?${permission}start=${start}`,
+            );
+            if (pageResult.values) {
+                result.push(...pageResult.values);
             }
-            if (segments.length > 0) {
-                q = `?${segments.join("&")}`;
+            isLastPage = !!pageResult.isLastPage;
```
Comment (AlexTugarev) [inline components/server/src/bitbucket-server/bitbucket-server-api.ts:303]: TODO: `isLastPage` should default to `true` if `pageResult.isLastPage` is undefined.

Comment (AlexTugarev) [reply components/server/src/bitbucket-server/bitbucket-server-api.ts:303]: going to change to: `isLastPage = typeof === "undefined" || !!pageResult.isLastPage; // test for "undefined" is a fuse`

**Review by filiptronicek — APPROVED** [review]
Code looks good

**Review by svenefftinge — APPROVED** [review]

**Pull request merged**